scipy/stats/_distn_infrastructure.py

```diff
@@ -2339,15 +2339,17 @@ def integ(x):
     def expect(self, func=None, args=(), loc=0, scale=1, lb=None, ub=None,
                conditional=False, **kwds):
         """Calculate expected value of a function with respect to the
-        distribution.
+        distribution by numeric integration.
 
         The expected value of a function ``f(x)`` with respect to a
         distribution ``dist`` is defined as::
 
                     ubound
-            E[x] = Integral(f(x) * dist.pdf(x))
+            E[f(x)] = Integral(f(x) * dist.pdf(x)),
```

Comment:
The description ist not very precise, it is the expected value of f(X) over the set {lb < X < lu} for a random variable X with the given distribution.

Reply:
Thanks for pointing this out. I agree this is a somewhat unclear description of an expected value calculation. I made this change on the push I just made. 

@@ -2348,7 +2348,8 @@ class rv_continuous(rv_generic):
             E[f(x)] = Integral(f(x) * dist.pdf(x)),
                     lbound
 
-        where ubound and lbound are the arguments ub and lb respectively. 
+        where ubound and lbound are the arguments ub and lb respectively
+        and `x` has the `dist.pdf(x)` distribution. 

@chrisb83, Does this change address your comment?

Reply:
formally I'm usually thinking about this in terms of multiplication with an indicator function 
`I{lb < X < lu}` 
(capital I or 1) that limits the nonzero range of the function to the integration interval, i.e.

`E[f(x) * I{lb < X < ub}] = \Integral_{lb}^{ub} (f(x) * dist.pdf(x)) dx`

and conditional `E[f(x) | lb < X < lu]`

Reply:
I'm not sure how formal or explicit to explain this

maybe add a sentence similar to your descriptions, e.g.
"The bounded integral forms the expectation of the function f(x) with respect to the given distribution of x if f(x) equal 0 outside the interval {lb <= X <= ub}"
If the bounds correspond to the support of the distribution, e.g. -inf and inf in the normal case, then the integral is the unrestricted expectation of f(x).

I like to think in terms of indicator function, because the expectation of the indicator function is just the probability that x is in the interval. But I doubt explaining it in terms of indicator function is helpful for the average user.

aside: In some cases specifying bounds can be helpful for integrate.quad to work better, In this case the truncation error will be small if there is only a tiny probability outside the interval. i.e bounds as numerical device without theoretical meaning.

Reply:
@ilayn, I like the suggestion and it sounds like @josef-pkt is on the same page. To avoid confusion I'll avoid the the of the overloaded word "normal" and use the word "usual"

to be explicit here is the diff/hunk: 

diff --git a/scipy/stats/_distn_infrastructure.py b/scipy/stats/_distn_infrastructure.py
index 8ffad6d0d..254e1bf08 100644
--- a/scipy/stats/_distn_infrastructure.py
+++ b/scipy/stats/_distn_infrastructure.py
@@ -2344,12 +2344,17 @@ class rv_continuous(rv_generic):
         The expected value of a function ``f(x)`` with respect to a
         distribution ``dist`` is defined as::
 
-                    ubound
+                    ub
             E[f(x)] = Integral(f(x) * dist.pdf(x)),
-                    lbound
+                    lb
 
-        where ubound and lbound are the arguments ub and lb respectively
-        and `x` has the `dist.pdf(x)` distribution. 
+        where `ub` and `lb` are arguments and `x` has the `dist.pdf(x)` 
+        distribution. If the bounds `lb` and `ub` correspond to the 
+        support of the distribution, e.g. `[-inf, inf]` in the default 
+        case, then the integral is the unrestricted expectation of `f(x)`. 
+        Also, the function `f(x)` may be defined such that `f(x)` is `0` 
+        outside a finite interval in which case the expectation is 
+        calculated within the finite range `[lb, ub]`. 
Any further objections?

Reply:
sounds good to me